src/entity/category/category_name.go

```diff
@@ -0,0 +1,15 @@
+package category
+
+import custerr "github.com/ha-zu/learn-clean-dev/src/entity/customerror"
+
+type CategoryName string
+
+func NewCategoryName(name CategoryName) error {
```

Comment:
これも
```suggestion
func NewCategoryName(name string) (CategoryName, error) {
```
このようなインターフェースになります。